File: server/src/main/java/org/ntcrckr/ppidpo/server/service/ServerController.java
package org.ntcrckr.ppidpo.server.service;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ServerController {

    CurrencyRateProvider currencyRateProvider;

    public ServerController(CurrencyRateProvider currencyRateProvider) {
        this.currencyRateProvider = currencyRateProvider;
    }

    @GetMapping("/currencyRate")
    public Double getCurrencyRate() {
        return currencyRateProvider.getCurrent();
    }
}

File: server/src/main/kotlin/org/ntcrckr/ppidpo2/rest/ServerController.kt
package org.ntcrckr.ppidpo2.rest

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class ServerController(
    private val currencyRateProvider: CurrencyRateProvider,
) {
    @GetMapping("/currencyRate")
    fun currencyRate(): Double = currencyRateProvider.getCurrentRate()
}